plugins/com.google.cloud.tools.eclipse.usagetracker/src/com/google/cloud/tools/eclipse/usagetracker/AnalyticsPingManager.java

```diff
@@ -259,7 +263,7 @@ public void showOptInDialog(final Shell parentShell) {
       display.syncExec(new Runnable() {
         @Override
         public void run() {
-          new OptInDialog(findShell(parentShell)).open();
+          new OptInDialog(findShell(parentShell), dialogOpenStatus).open();
```

Comment:
It seems weird to pass this in to the dialog: why is it the dialog's responsibility to manage itself?  Is this for testing? Why not:
```
if (dialogOpenStatus.compareAndSet(false, true)) {
   display.syncExec(…);
}
```
While we're here, the `show` in `showOptInDialog` doesn't convey that it is conditional.  `verifyOptIn` or `verifyConsent` or something?